src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/FilteringBindingProvider.cs

```diff
@@ -26,7 +33,16 @@ public Task<IBinding> TryCreateAsync(BindingProviderContext context)
                 throw new ArgumentNullException("context");
             }
             Type parametertype = context.Parameter.ParameterType;
-            if (!_predicate(parametertype))
+
+            var attr = context.Parameter.GetCustomAttribute<TAttribute>();
+
+            var cloner = new AttributeCloner<TAttribute>(attr, _nameResolver);
+            var attrNameResolved = cloner.GetNameResolvedAttribute();
+
+            // This may do validation and throw too. 
+            bool canBind = _predicate(attrNameResolved, parametertype);
+
+            if (!canBind)
```

Comment:
The method name 'AddFilter' does not really convey that to me. I'm guessing most people will read this as 'there's a list of filters that get run, and I'm adding to it'.

---
In reply to: [63771789](https://github.com/Azure/azure-webjobs-sdk/pull/726#discussion_r63771789) [](ancestors = 63771789,63770416)